python/persistenceDrivenCompression.py

```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env python
+
+from paraview.simple import *
+
+# create a new 'TTK TopologicalCompressionReader'
+naturalImage_zfp50ttk = TTKTopologicalCompressionReader(
+    FileName="naturalImage_zfp50.ttk"
+)
+
+# create a new 'XML Image Data Reader'
+naturalImage_originalvti = XMLImageDataReader(FileName=["naturalImage_original.vti"])
+naturalImage_originalvti.PointArrayStatus = ["PNGImage"]
+naturalImage_originalvti.TimeArray = "None"
+
+# create a new 'TTK TopologicalCompressionReader'
+naturalImage_persistence10ttk = TTKTopologicalCompressionReader(
+    FileName="naturalImage_persistence10.ttk",
+)
+
+# create a new 'TTK TopologicalCompressionReader'
+naturalImage_persistence10_zfp50ttk = TTKTopologicalCompressionReader(
+    FileName="naturalImage_persistence10_zfp50.ttk",
+)
```

Comment:
I do not believe this script is actually executing anything.
(without triggering any pipeline update, nothing will happen)

I think you want to store the 2 compressed images in a more common file format (i.e. VTI).
then this example will illustrate how to convert an image compressed with TTK to a more common file format.
for instance:
```
SaveData("uncompressed_naturalImage_persistence10.vti", naturalImage_persistence10ttk)
SaveData("uncompressed_naturalImage_persistence10_zfp50", naturalImage_persistence10_zfp50ttk)

Reply:
I've had a second thought about this script.
I think, to exemplify both the compression and the decompression, we should proceed as follows:

1. Read the file naturalImage_original.vti
2. Compress it with a persistence of 10% and zfp 50 and save in .ttk file
3. Decomress the generated ttk file (with the TTKTopologicalCompressionReader)
4. Save it as a uncompressed vti file.

Reply:
Something like that?

```python
#!/usr/bin/env python

from paraview import simple

# read input VTI with 'XML Image Data Reader'
naturalImage = simple.XMLImageDataReader(FileName=["naturalImage_original.vti"])
naturalImage.PointArrayStatus = ["PNGImage"]

# compress & save to TTK Topological Compression format
simple.SaveData(
    "naturalImage_persistence10_zfp50.ttk",
    proxy=naturalImage,
    ScalarField=["POINTS", "PNGImage"],
    Topologicallosspersistencepercentage=10,  # Topological loss
    ZFPRelativeErrorToleranceextra=50,  # ZFP Relative Error Tolerance
)

# read the compressed file with 'TTK TopologicalCompressionReader'
naturalImage_compressed = simple.TTKTopologicalCompressionReader(
    FileName="naturalImage_persistence10_zfp50.ttk",
)

# write compressed data-set to VTI
simple.SaveData(
    "uncompressed_naturalImage_persistence10_zfp50.vti",
    naturalImage_compressed,
)
```

Reply:
absolutely!
(I don't believe the line `naturalImage.PointArrayStatus = ["PNGImage"]` is necessary though)

Reply:
I would also simply change to `from paraview.simple import *` (to make the rest of the script simpler)

Reply:
Should be good now